msdsl/model.py

```diff
@@ -189,25 +189,37 @@ def add_assignment(self, assignment: Assignment):
 
         return assignment.signal
 
-    def immediate_assign(self, signal: Union[Signal, str], expr: ModelExpr):
+    def immediate_assign(
+            self, signal: Union[Signal, str], expr: ModelExpr, range_=None,
+            width=None, exponent=None, check_format=True
+    ):
         """
         Alias for set_this_cycle.
         """
 
-        return self.set_this_cycle(signal=signal, expr=expr)
+        return self.set_this_cycle(
+            signal=signal, expr=expr, range_=range_, width=width,
+            exponent=exponent, check_format=check_format
+        )
 
-    def bind_name(self, signal: Union[Signal, str], expr: ModelExpr, range_=None,
-                  width=None, exponent=None):
+    def bind_name(
+            self, signal: Union[Signal, str], expr: ModelExpr, range_=None,
+            width=None, exponent=None, check_format=True
+    ):
         """
         TODO: consider deprecating.
         Alias for set_this_cycle.
         """
 
-        return self.set_this_cycle(signal=signal, expr=expr, range_=range_,
-                                   width=width, exponent=exponent)
+        return self.set_this_cycle(
+            signal=signal, expr=expr, range_=range_, width=width,
+            exponent=exponent, check_format=check_format
+        )
 
-    def set_this_cycle(self, signal: Union[Signal, str], expr: ModelExpr, range_=None,
-                       width=None, exponent=None):
+    def set_this_cycle(
+            self, signal: Union[Signal, str], expr: ModelExpr, range_=None,
+            width=None, exponent=None, check_format=True
```

Comment:
Note the new ``check_format`` option.  It defaults to ``True``, which is almost always what you'll want.  Setting it ``False`` disables checks that would look at bit widths, ranges, etc.